src/zepben/auth/client/token_fetcher.py

```diff
@@ -150,18 +149,23 @@ def create_token_fetcher(conf_address: str, verify_certificates: bool = True, au
     with warnings.catch_warnings():
         if not verify_certificates:
             warnings.filterwarnings("ignore", category=InsecureRequestWarning)
-        response = requests.get(conf_address, verify=verify_certificates and (conf_ca_filename or True))
+        try:
+            response = requests.get(conf_address, verify=verify_certificates and (conf_ca_filename or True))
+        except Exception as e:
+            warnings.warn(str(e))
+            warnings.warn("If RemoteDisconnected, check if eas_https in config is set to the same as the target EAS server. This process may hang indefinetly.")
+            raise ConnectionError("Check if eas_https in config is set to the same as the target EAS server")
```

Comment:
it doesn't make sense to directly reference "eas_https" here, as any code could be using this library and run into this issue - maybe a more generic error message like "Are you trying to connect to a HTTPS server with HTTP?"